mesonbuild/dependencies/base.py

```diff
@@ -143,6 +144,22 @@ def get_pkgconfig_variable(self, variable_name, kwargs):
     def get_configtool_variable(self, variable_name):
         raise DependencyException('{!r} is not a config-tool dependency'.format(self.name))
 
+    def get_partial_dependency(self, *, compile_args=False, link_args=False,
+                               links=False, includes=False, sources=False):
+        """Create a new dependency that contains part of the parent dependency.
+
+        The following options can be inherited:
+            links -- all link_with arguemnts
+            includes -- all include_directory and -I/-isystem calls
+            sources -- any source, header, or generated sources
+            compile_args -- any compile args
+            link_args -- any link args
+
+        Additionally the new dependency will have the version parameter of it's
+        parent (if any) and the requested values of any dependencies will be
+        added as well.
+        """
```

Comment:
If the body of this method is empty, it should have `pass` for clarity. Or, if this should not be called ever, `raise RuntimeError('Unreachable code in partial_dependency called.')`